arbitrator/prover/src/machine.rs

```diff
@@ -2315,6 +2333,21 @@ impl Machine {
                         panic!("Should never ever get here");
                     }
                 }
+            } else if matches!(next_inst.opcode, Opcode::ReadHotShotCommitment) {
+                let ptr = self.value_stack.get(0).unwrap().assume_u32();
+                if let Some(mut idx) = usize::try_from(ptr).ok().filter(|x| x % 32 == 0) {
+                    idx /= Memory::LEAF_SIZE;
+                    let prev_data = module.memory.get_leaf_data(idx);
+                    data.extend(prev_data);
+                    data.extend(mem_merkle.prove(idx).unwrap_or_default());
```

Comment:
@ImJeremyHe what is the this mem_merkle?

Reply:
The memory(Vec<u8>) of this module. This is to prove the existence of the variable in the memory

Reply:
got it